doc/rose-rug-advanced-tutorials-macro.html

```diff
@@ -589,6 +589,21 @@ <h3 class="alwayshidden">Example Metadata Option</h3>
             </div>
 
             <div class="slide">
+              <h2 id="reporter-macros">Reporter Macros</h2>
+              <p>There are also reporter macros which can be used
+              where you need to output some information about a
+              configuration. A reporter macro takes the same form
+              as validator and transform macros but does not require
+              a return value.</p>
+
+              <pre class="prettyprint">
+    def report(self, config, meta_config=None):
```

Comment:
Please make the example here part of the tutorial itself so someone can work through this against the rest of the tutorial. At present this is just a copy of what's in the API doc and not relevant to the tutorial itself - there's no snowflakes here. I recommend something like "planet.PlanetReporter" to write the planet name to a file.